tsl/src/continuous_aggs/options.c

```diff
@@ -194,28 +194,37 @@ cagg_get_compression_params(ContinuousAgg *agg, Hypertable *mat_ht)
 	return defelems;
 }
 
-/* enable/disable compression on continuous aggregate */
+/* forwards compression related changes via an alter statement to the underlying HT */
 static void
-cagg_alter_compression(ContinuousAgg *agg, Hypertable *mat_ht, bool compress_enable)
+cagg_alter_compression(ContinuousAgg *agg, Hypertable *mat_ht, List *compress_defelems)
 {
-	List *defelems = NIL;
 	Assert(mat_ht != NULL);
-	if (compress_enable)
-		defelems = cagg_get_compression_params(agg, mat_ht);
+	WithClauseResult *with_clause_options =
+		ts_compress_hypertable_set_clause_parse(compress_defelems);
 
-	DefElem *enable = makeDefElemExtended("timescaledb",
-										  "compress",
-										  compress_enable ? (Node *) makeString("true") :
-															(Node *) makeString("false"),
-										  DEFELEM_UNSPEC,
-										  -1);
-	defelems = lappend(defelems, enable);
+	if (with_clause_options[CompressEnabled].parsed)
+	{
+		List *default_compress_defelems = cagg_get_compression_params(agg, mat_ht);
+		WithClauseResult *default_with_clause_options =
+			ts_compress_hypertable_set_clause_parse(default_compress_defelems);
+		// merge defaults if there's any
```

Comment:
```suggestion
		/* Merge defaults if there's any. */
```

Let's follow the Postgres comment style.